### DISCUSSION TITLE
Get the Advertised product report for 90 days but the maximum date range is 31 days.

### DISCUSSION THREAD

**Discussion by Karthick2299** [amzn/ads-advanced-tools-docs#157] [Q&A] [answered]
Hi, 
I need to get the advertised product report on sponsored products , I already get this report with maximum date range of 31 days only. I want to get this report by 90 days. 
Is this possible to get the 90 days of advertised product report ?

Thanks! in Advance

**Comment by elizafarley** [comment, marked as answer]
Hi @Karthick2299, as you pointed out, each report request can only have a maximum of 31 days, so there's no way to get 90 days of data in a single call. You would need to make three calls: one for days 1 - 30, the second with days 30 - 60, and the third with days 60 - 90.

Also, please note that while you can make a request up to 95 days in the past, data is only returned for the last 65 days. This is a currently a [known issue](https://advertising.amazon.com/API/docs/en-us/support/known-issues#incomplete-data-retention-for-sponsored-products-reports-in-version-3).

**Comment by Karthick2299** [comment]
Hi @elizafarley, The response you given is very helpful for our team to know more about the Amazon ads api.
The link you given about `known issue` is not working. it shows not found(404) on Github.
could you look into it ?
Thanks :)

**Reply by elizafarley** [reply]
Sorry about that! Link should be working now.

**Reply by Karthick2299** [reply]
Thanks @elizafarley

**Comment by Karthick2299** [comment]
Hi @elizafarley, I believe the issue of obtaining a 95-day report has been resolved. 
I can now successfully generate a report spanning 95 days. 
Is the problem truly resolved or am i getting the wrong report?

I generated report before 95 days of today(2023-10-13). 
The report generated `startDate` - `2023-07-11` and `endDate` - `2023-07-31`

Thanks!

**Reply by elizafarley** [reply]
You will be able to request a report 95 days in the past, however I don't believe there will actually be any data in it.

**Reply by Karthick2299** [reply]
There is a data in it. See the screenshot
![Screenshot 2023-10-16 at 12 16 46 PM](https://github.com/amzn/ads-advanced-tools-docs/assets/106722215/d331ed52-6f55-47a8-a9ce-b8da99ecb02b)
